mods/page_save.js

```diff
@@ -36,11 +36,49 @@ function PageSave(options) {
     };
 }
 
-PageSave.prototype._getRevInfo = function(restbase, req) {
+PageSave.prototype._getStartTimestamp = function(req) {
+    if (req.headers && req.headers['if-match']) {
+        var etag = rbUtil.parseETag(req.headers['if-match']);
+        if (!etag || !TimeUuid.test(etag.tid) || !/^(?:[0-9]+)$/.test(etag.rev)) {
+            throw new rbUtil.HTTPError({
+                status: 400,
+                body: {
+                    type: 'invalid_request',
+                    title: 'Bad ETag in If-Match',
+                    description: 'The supplied base_etag is invalid'
+                }
+            });
+        } else {
+            return TimeUuid.fromString(etag.tid).getDate().toISOString();
+        }
+    }
+    return undefined;
+};
+
+PageSave.prototype._getBaseRevision = function(req) {
```

Comment:
@d00rman The if-match header contains a latest revision, base revision is in base_etag.